{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Human1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrM</td>\n",
       "      <td>73</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>24.00</td>\n",
       "      <td>LowGQX</td>\n",
       "      <td>SNVSB=0.0;SNVHPOL=3</td>\n",
       "      <td>GT:GQ:GQX:DP:DPF:AD</td>\n",
       "      <td>1/1:5:5:2:0:0,2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrM</td>\n",
       "      <td>150</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>91.00</td>\n",
       "      <td>LowGQX</td>\n",
       "      <td>SNVSB=-9.2;SNVHPOL=2</td>\n",
       "      <td>GT:GQ:GQX:DP:DPF:AD</td>\n",
       "      <td>1/1:10:10:4:0:0,4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrM</td>\n",
       "      <td>195</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>44.00</td>\n",
       "      <td>LowGQX</td>\n",
       "      <td>SNVSB=-8.9;SNVHPOL=2</td>\n",
       "      <td>GT:GQ:GQX:DP:DPF:AD</td>\n",
       "      <td>1/1:5:5:2:0:0,2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrM</td>\n",
       "      <td>302</td>\n",
       "      <td>.</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>75.00</td>\n",
       "      <td>LowGQX</td>\n",
       "      <td>CIGAR=1M1D;RU=C;REFREP=8;IDREP=7</td>\n",
       "      <td>GT:GQ:GQX:DPI:AD</td>\n",
       "      <td>1/1:12:10:4:0,4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chrM</td>\n",
       "      <td>410</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>59.00</td>\n",
       "      <td>LowGQX</td>\n",
       "      <td>SNVSB=0.0;SNVHPOL=4</td>\n",
       "      <td>GT:GQ:GQX:DP:DPF:AD</td>\n",
       "      <td>1/1:10:10:4:0:0,4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0 #CHROM  POS ID REF ALT   QUAL  FILTER                              INFO  \\\n",
       "1   chrM   73  .   G   A  24.00  LowGQX               SNVSB=0.0;SNVHPOL=3   \n",
       "2   chrM  150  .   T   C  91.00  LowGQX              SNVSB=-9.2;SNVHPOL=2   \n",
       "3   chrM  195  .   C   T  44.00  LowGQX              SNVSB=-8.9;SNVHPOL=2   \n",
       "4   chrM  302  .  AC   A  75.00  LowGQX  CIGAR=1M1D;RU=C;REFREP=8;IDREP=7   \n",
       "5   chrM  410  .   A   T  59.00  LowGQX               SNVSB=0.0;SNVHPOL=4   \n",
       "\n",
       "0               FORMAT             Human1  \n",
       "1  GT:GQ:GQX:DP:DPF:AD    1/1:5:5:2:0:0,2  \n",
       "2  GT:GQ:GQX:DP:DPF:AD  1/1:10:10:4:0:0,4  \n",
       "3  GT:GQ:GQX:DP:DPF:AD    1/1:5:5:2:0:0,2  \n",
       "4     GT:GQ:GQX:DPI:AD    1/1:12:10:4:0,4  \n",
       "5  GT:GQ:GQX:DP:DPF:AD  1/1:10:10:4:0:0,4  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "input_file = \"test.vcf\"\n",
    "\n",
    "big_file = open(\"SQAHuman01_S1.vcf\", \"r\")\n",
    "found = False\n",
    "array = []\n",
    "for line in big_file:\n",
    "    if line.startswith(\"#CHROM\"):\n",
    "        found = True\n",
    "    if found:\n",
    "        temp = line.replace('\\n', '').split('\\t')\n",
    "        array.append(temp)\n",
    "        \n",
    "#print array[0:5]\n",
    "df_test = pd.DataFrame.from_records(array)\n",
    "df_test.columns = df_test.iloc[0]\n",
    "df_test = df_test[1:]\n",
    "df_test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Human1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>chrM</td>\n",
       "      <td>16306</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>203.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SNVSB=-24.7;SNVHPOL=2</td>\n",
       "      <td>GT:GQ:GQX:DP:DPF:AD</td>\n",
       "      <td>1/1:39:39:14:1:0,14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>chrM</td>\n",
       "      <td>16322</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>180.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SNVSB=-19.3;SNVHPOL=2</td>\n",
       "      <td>GT:GQ:GQX:DP:DPF:AD</td>\n",
       "      <td>1/1:33:33:12:1:0,12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1071</th>\n",
       "      <td>chr1</td>\n",
       "      <td>38868531</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>44.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SNVSB=-8.2;SNVHPOL=2</td>\n",
       "      <td>GT:GQ:GQX:DP:DPF:AD</td>\n",
       "      <td>0/1:40:38:3:0:1,2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1271</th>\n",
       "      <td>chr1</td>\n",
       "      <td>46434279</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>41.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SNVSB=-7.5;SNVHPOL=3;CSQT=MAST2|ENST0000036129...</td>\n",
       "      <td>GT:GQ:GQX:DP:DPF:AD</td>\n",
       "      <td>0/1:39:37:3:0:1,2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2902</th>\n",
       "      <td>chr1</td>\n",
       "      <td>90035989</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SNVSB=-8.5;SNVHPOL=3;CSQT=LRRC8B|ENST000003309...</td>\n",
       "      <td>GT:GQ:GQX:DP:DPF:AD</td>\n",
       "      <td>0/1:36:34:3:0:1,2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     #CHROM       POS ID REF ALT   QUAL FILTER  \\\n",
       "50     chrM     16306  .   T   C  203.0   PASS   \n",
       "51     chrM     16322  .   T   C  180.0   PASS   \n",
       "1071   chr1  38868531  .   C   A   44.0   PASS   \n",
       "1271   chr1  46434279  .   C   T   41.0   PASS   \n",
       "2902   chr1  90035989  .   G   A   40.0   PASS   \n",
       "\n",
       "                                                   INFO               FORMAT  \\\n",
       "50                                SNVSB=-24.7;SNVHPOL=2  GT:GQ:GQX:DP:DPF:AD   \n",
       "51                                SNVSB=-19.3;SNVHPOL=2  GT:GQ:GQX:DP:DPF:AD   \n",
       "1071                               SNVSB=-8.2;SNVHPOL=2  GT:GQ:GQX:DP:DPF:AD   \n",
       "1271  SNVSB=-7.5;SNVHPOL=3;CSQT=MAST2|ENST0000036129...  GT:GQ:GQX:DP:DPF:AD   \n",
       "2902  SNVSB=-8.5;SNVHPOL=3;CSQT=LRRC8B|ENST000003309...  GT:GQ:GQX:DP:DPF:AD   \n",
       "\n",
       "                   Human1  \n",
       "50    1/1:39:39:14:1:0,14  \n",
       "51    1/1:33:33:12:1:0,12  \n",
       "1071    0/1:40:38:3:0:1,2  \n",
       "1271    0/1:39:37:3:0:1,2  \n",
       "2902    0/1:36:34:3:0:1,2  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_table(input_file)\n",
    "df_pass = df[df[\"FILTER\"] == \"PASS\"]\n",
    "df_pass.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[30, 30, 30, 30, 30, 31, 31, 31, 31, 31, 31, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 33, 33, 33, 33, 33, 33, 33, 33, 33, 33, 33, 33, 33, 33, 33, 34, 34, 34, 34, 34, 34, 34, 34, 34, 34, 34, 35, 35, 35, 35, 35, 35, 35, 35, 35, 36, 36, 36, 36, 36, 36, 36, 36, 36, 36, 36, 36, 36, 36, 36, 36, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 38, 38, 38, 38, 38, 38, 38, 38, 39, 39, 39, 39, 43, 45, 45, 45, 45, 46, 47, 47, 47, 47, 48, 48, 48, 48, 49, 49, 50, 51, 51, 51, 53, 54, 56, 57, 57, 57, 58, 58, 59, 61, 61, 62, 62, 63, 63, 65, 67, 68, 69, 69, 72, 72, 72, 73, 73, 73, 73, 73, 74, 74, 75, 75, 75, 77, 78, 78, 81, 84, 85, 89, 90, 91, 92, 95, 98, 100, 100, 114, 116, 118, 121, 129, 130, 141, 145, 147, 152, 153, 160, 169, 175, 180, 238, 477]\n"
     ]
    }
   ],
   "source": [
    "GQX_loc = []\n",
    "score = []\n",
    "res = []\n",
    "for entry in df_pass[\"FORMAT\"]:\n",
    "    temp = entry.split(\":\")\n",
    "    loc = temp.index(\"GQX\")\n",
    "    GQX_loc.append(loc)\n",
    "\n",
    "for entry in df_pass[\"Human1\"]:\n",
    "    score.append(entry)\n",
    "\n",
    "for i in xrange(len(score)):\n",
    "    temp = score[i].split(\":\")\n",
    "    loc = GQX_loc[i]\n",
    "    res.append(temp[loc])\n",
    "    \n",
    "print sorted(map(int, res))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.2.0\n"
     ]
    }
   ],
   "source": [
    "import allel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[u'calldata/AD',\n",
       " u'calldata/DP',\n",
       " u'calldata/DPF',\n",
       " u'calldata/DPI',\n",
       " u'calldata/GQ',\n",
       " u'calldata/GQX',\n",
       " u'calldata/GT',\n",
       " 'samples',\n",
       " u'variants/AA',\n",
       " u'variants/AF1000G',\n",
       " u'variants/ALT',\n",
       " u'variants/BLOCKAVG_min30p3a',\n",
       " 'variants/CHROM',\n",
       " u'variants/CIGAR',\n",
       " u'variants/CSQR',\n",
       " u'variants/CSQT',\n",
       " u'variants/END',\n",
       " u'variants/EVS',\n",
       " u'variants/FILTER_HighDPFRatio',\n",
       " u'variants/FILTER_HighDepth',\n",
       " u'variants/FILTER_HighSNVSB',\n",
       " u'variants/FILTER_IndelConflict',\n",
       " u'variants/FILTER_LowGQX',\n",
       " u'variants/FILTER_PASS',\n",
       " u'variants/FILTER_SiteConflict',\n",
       " u'variants/GMAF',\n",
       " u'variants/ID',\n",
       " u'variants/IDREP',\n",
       " 'variants/POS',\n",
       " u'variants/QUAL',\n",
       " 'variants/REF',\n",
       " u'variants/REFREP',\n",
       " u'variants/RU',\n",
       " u'variants/SNVHPOL',\n",
       " u'variants/SNVSB',\n",
       " 'variants/altlen',\n",
       " u'variants/clinvar',\n",
       " u'variants/cosmic',\n",
       " 'variants/is_snp',\n",
       " 'variants/numalt',\n",
       " u'variants/phastCons']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_set = allel.read_vcf('SQAHuman01_S1.vcf', fields='*')\n",
    "\n",
    "sorted(test_set.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([False, False, False, ..., False, False, False])"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gqx = test_set['calldata/GQX']\n",
    "filter_p = test_set['variants/FILTER_PASS']\n",
    "filter_p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[30,\n",
       " 30,\n",
       " 30,\n",
       " 30,\n",
       " 30,\n",
       " 31,\n",
       " 31,\n",
       " 31,\n",
       " 31,\n",
       " 31,\n",
       " 31,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 32,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 33,\n",
       " 34,\n",
       " 34,\n",
       " 34,\n",
       " 34,\n",
       " 34,\n",
       " 34,\n",
       " 34,\n",
       " 34,\n",
       " 34,\n",
       " 34,\n",
       " 34,\n",
       " 35,\n",
       " 35,\n",
       " 35,\n",
       " 35,\n",
       " 35,\n",
       " 35,\n",
       " 35,\n",
       " 35,\n",
       " 35,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 36,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 37,\n",
       " 38,\n",
       " 38,\n",
       " 38,\n",
       " 38,\n",
       " 38,\n",
       " 38,\n",
       " 38,\n",
       " 38,\n",
       " 39,\n",
       " 39,\n",
       " 39,\n",
       " 39,\n",
       " 43,\n",
       " 45,\n",
       " 45,\n",
       " 45,\n",
       " 45,\n",
       " 46,\n",
       " 47,\n",
       " 47,\n",
       " 47,\n",
       " 47,\n",
       " 48,\n",
       " 48,\n",
       " 48,\n",
       " 48,\n",
       " 49,\n",
       " 49,\n",
       " 50,\n",
       " 51,\n",
       " 51,\n",
       " 51,\n",
       " 53,\n",
       " 54,\n",
       " 56,\n",
       " 57,\n",
       " 57,\n",
       " 57,\n",
       " 58,\n",
       " 58,\n",
       " 59,\n",
       " 61,\n",
       " 61,\n",
       " 62,\n",
       " 62,\n",
       " 63,\n",
       " 63,\n",
       " 65,\n",
       " 67,\n",
       " 68,\n",
       " 69,\n",
       " 69,\n",
       " 72,\n",
       " 72,\n",
       " 72,\n",
       " 73,\n",
       " 73,\n",
       " 73,\n",
       " 73,\n",
       " 73,\n",
       " 74,\n",
       " 74,\n",
       " 75,\n",
       " 75,\n",
       " 75,\n",
       " 77,\n",
       " 78,\n",
       " 78,\n",
       " 81,\n",
       " 84,\n",
       " 85,\n",
       " 89,\n",
       " 90,\n",
       " 91,\n",
       " 92,\n",
       " 95,\n",
       " 98,\n",
       " 100,\n",
       " 100,\n",
       " 114,\n",
       " 116,\n",
       " 118,\n",
       " 121,\n",
       " 129,\n",
       " 130,\n",
       " 141,\n",
       " 145,\n",
       " 147,\n",
       " 152,\n",
       " 153,\n",
       " 160,\n",
       " 169,\n",
       " 175,\n",
       " 180,\n",
       " 238,\n",
       " 477]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(filter_p)\n",
    "df['gqx'] = gqx\n",
    "df.columns = ['pass', 'gqx']\n",
    "df = df[df['pass'] == True]\n",
    "sorted(df['gqx'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
